File: library/src/main/java/com/ebr163/inspector/AbstractInspection.java
package com.ebr163.inspector;

import com.ebr163.inspector.rule.Rule;

import java.util.List;

/**
 * Created by Bakht
 * on 09.01.2018.
 */

public abstract class AbstractInspection<Type> implements Inspection<Type> {

    protected List<Rule<Type>> rules;

    protected AbstractInspection(List<Rule<Type>> rules){
        this.rules = rules;
    }

    @Override
    public void clear() {
        rules.clear();
    }
}

File: library/src/main/java/com/ebr163/inspector/AbstractInspection.kt
package com.ebr163.inspector

import com.ebr163.inspector.rule.Rule

/**
 * Created by Bakht
 * on 09.01.2018.
 */

abstract class AbstractInspection<Type> protected constructor(protected var rules: MutableList<Rule<Type>>) : Inspection<Type> {

    override fun clear() = rules.clear()
}
